msu/utils/globals.nut

```diff
@@ -127,3 +127,28 @@
 		_object = _object.get();
 	return typeof _object == "table" && "_release_hook_DO_NOT_delete_it_" in _object;
 }
+
+::MSU.deepClone <- function( _object )
+{
+	local ret;
+	switch (typeof _object)
+	{
+		case "table":
+			ret = {};
+			foreach (key, value in _object)
+			{
+				ret[key] <- ::MSU.deepClone(value);
```

Comment:
How does cloning of tables handle delegation? Ideally we'd want a deep clone of a table with tables which has delegation set up should return a setup with the exact same delegation setup.

Reply:
This is a good point and I agree